packages/block-library/src/image/image.js

```diff
@@ -591,25 +615,28 @@ export default function Image( {
 				which causes duplicated image upload. */ }
 			{ ! temporaryURL && controls }
 			{ img }
-			{ ( ! RichText.isEmpty( caption ) || isSelected ) && (
-				<RichText
-					className={ __experimentalGetElementClassName( 'caption' ) }
-					ref={ captionRef }
-					tagName="figcaption"
-					aria-label={ __( 'Image caption text' ) }
-					placeholder={ __( 'Add caption' ) }
-					value={ caption }
-					onChange={ ( value ) =>
-						setAttributes( { caption: value } )
-					}
-					inlineToolbar
-					__unstableOnSplitAtEnd={ () =>
-						insertBlocksAfter(
-							createBlock( getDefaultBlockName() )
-						)
-					}
-				/>
-			) }
+			{ showCaption &&
```

Comment:
Why is it possible to hide the caption when it's not empty? I think we should remove this additional state and purely look at the block attribute value. If the caption value is undefined (or null), the caption should be hidden. If the caption is a string (can be empty), show the caption. The caption button should modify the block to insert an empty string.

Reply:
It's from the initial approach to have a toggle add/remove caption, which it turns out I'll have to redo for [a11y reasons](https://github.com/WordPress/gutenberg/pull/44965#issuecomment-1280992622).